### ISSUE TITLE
conformance: Conformance tests need a way to test that a percentage of requests succeed or fail

### ISSUE THREAD

**Issue by youngnick** [kubernetes-sigs/gateway-api#1250]
**What would you like to be added**:
In #1243, I added some conformance testing around new BackendRef tests, but one case we couldn't catch is the following:
- There are two backendRefs in the HTTPRoute backendRef list, with equal weights
- One is invalid
- Half of the requests sent through the implementation must get a 500.

**Why this is needed**:
Part of the spec is weighted load balancing, having some tooling to allow us to test this as part of conformance will help a lot.

Downside: It's going to be tricky to write the tests to be not flaky. I suspect that we'll need to provide a target percentage value, add some wiggle room to it, and then send some number of requests. It seems likely to me that the details will matter a lot here.

May need to be a GEP, let's discuss here first.

**Comment by mikemorris** [comment]
This definitely feels like it would be flaky to test - would using filters to direct 100% of traffic to each of a valid and invalid backend for the same route be a viable alternative to this?

**Comment by youngnick** [comment]
Well, in the spec for backendRef, the above behavior is mandated (that is, if there are multiple backends, and one is invalid, then the invalid one should instead produce an equivalent percentage of 500 errors), so ideally we should have a way to test implementations do that in conformance.

That's why I put a target percentage value, and some wiggle room (maybe 50% plus minus 5% or something for the two-backend case).

Having filters direct traffic doesn't meet the spec item that we're trying to test, unfortunately.

**Comment by youngnick** [comment]
I agree that we should create another issue for the test Mike described.

**Comment by mikemorris** [comment]
Well, looking at [HTTPRouteFilter](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.HTTPRouteFilter) it appears the functionality I was envisioning doesn't really exist (so I'm not opening an issue to test it hah) - [HTTPRequestMirrorFilter](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.HTTPRequestMirrorFilter) is somewhat similar but would ignore the actual configured `backendRefs` on the HTTPRoute, I was thinking of something like an HTTPRequestLabelSelectorFilter configuring https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/ to filter backendRefs by metadata.
